avm/res/compute/gallery/main.bicep

```diff
@@ -220,4 +236,63 @@ type roleAssignmentType = {
 
   @sys.description('Optional. The Resource Id of the delegated managed identity resource.')
   delegatedManagedIdentityResourceId: string?
-}[]?
+}[]
+
+import { identifierType, purchasePlanType, resourceRangeType } from './image/main.bicep'
+type imageType = {
+  @sys.description('Required. Name of the image definition.')
+  @minLength(1)
+  @maxLength(80)
+  name: string
+
+  @sys.description('Optional. The description of this gallery image definition resource. This property is updatable.')
+  description: string?
+
+  @sys.description('Required. This property allows you to specify the type of the OS that is included in the disk when creating a VM from a managed image.')
+  osType: ('Linux' | 'Windows')
+
+  @sys.description('Required. This property allows the user to specify the state of the OS of the image.')
+  osState: ('Generalized' | 'Specialized')
+
+  @sys.description('Required. This is the gallery image definition identifier.')
+  identifier: identifierType
+
+  @sys.description('Optional. Describes the resource range (1-128 CPU cores). Defaults to min=1, max=4.')
```

Comment:
Question: You have the range so neatly written here. I wonder - should we add the min & max directly as annotations to the property in the type instead? Then we get direct language support.
Same question for the memory parameter

Reply:
Ah, now I see 2 different properties use the same UDT. I'd almost suggest to split the to enable the rules as subjectively it's always better to fail before trying to even deploy, but you call 😄 Feel free to resolve